polars/polars-core/src/frame/groupby/aggregations.rs

```diff
@@ -679,62 +680,101 @@ impl<T: PolarsObject> AggList for ObjectChunked<T> {
+{
+    pub(crate) fn agg_quantile(
         &self,
-        _groups: &[(u32, Vec<u32>)],
-        _quantile: f64,
-        _interpol: QuantileInterpolOptions,
+        groups: &[(u32, Vec<u32>)],
+        quantile: f64,
+        interpol: QuantileInterpolOptions,
     ) -> Option<Series> {
-        None
+        let invalid_quantile = !(0.0..=1.0).contains(&quantile);
+        let ca = &self.0;
+        agg_helper::<T, _>(groups, |(_first, idx)| {
+            debug_assert!(idx.len() <= ca.len());
+            if idx.is_empty() | invalid_quantile {
+                return None;
+            }
+
+            let take = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
+            take.into_series()
+                .quantile_as_series(quantile, interpol)
+                .unwrap() // Validity of quantile checked
+                .unpack::<T>()
+                .unwrap()
+                .get(0)
+        })
     }
 
-    fn agg_median(&self, _groups: &[(u32, Vec<u32>)]) -> Option<Series> {
-        None
+    pub(crate) fn agg_median(&self, groups: &[(u32, Vec<u32>)]) -> Option<Series> {
+        let ca = &self.0;
+        agg_helper::<T, _>(groups, |(_first, idx)| {
+            debug_assert!(idx.len() <= ca.len());
+            if idx.is_empty() {
+                return None;
+            }
+
+            let take = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
+            take.into_series()
+                .median_as_series()
+                .unpack::<T>()
+                .unwrap()
+                .get(0)
+        })
     }
 }
 
-impl<T> AggQuantile for ChunkedArray<T>
+impl<T> ChunkedArray<T>
 where
-    T: PolarsNumericType + Sync,
+    T: PolarsIntegerType + Sync,
+    ChunkedArray<T>: IntoSeries,
     T::Native: PartialOrd + Num + NumCast + Zero + Simd + std::iter::Sum<T::Native>,
     <T::Native as Simd>::Simd: std::ops::Add<Output = <T::Native as Simd>::Simd>
         + arrow::compute::aggregate::Sum<T::Native>
         + arrow::compute::aggregate::SimdOrd<T::Native>,
-    ChunkedArray<T>: IntoSeries,
 {
-    fn agg_quantile(
+    pub(crate) fn agg_quantile(
         &self,
         groups: &[(u32, Vec<u32>)],
         quantile: f64,
         interpol: QuantileInterpolOptions,
     ) -> Option<Series> {
-        agg_helper::<T, _>(groups, |(_first, idx)| {
-            if idx.is_empty() {
+        let invalid_quantile = !(0.0..=1.0).contains(&quantile);
+
+        agg_helper::<Float64Type, _>(groups, |(_first, idx)| {
+            if idx.is_empty() | invalid_quantile {
                 return None;
             }
 
-            let group_vals = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
-            group_vals.quantile(quantile, interpol).unwrap()
+            let take = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
+            take.into_series()
+                .quantile_as_series(quantile, interpol)
```

Comment:
Is this cast to Series necessary? I thought quantile_as_series was implemented for `ChunkedArray`.